albumentations/augmentations/transforms.py

```diff
@@ -792,8 +820,41 @@ def get_params_dependent_on_targets(self, params):
 
         holes = []
         for _n in range(random.randint(self.min_holes, self.max_holes)):
-            hole_height = random.randint(self.min_height, self.max_height)
-            hole_width = random.randint(self.min_width, self.max_width)
+            if all(
+                [
+                    isinstance(self.min_height, int),
+                    isinstance(self.min_width, int),
+                    isinstance(self.max_height, int),
+                    isinstance(self.max_width, int),
+                ]
+            ):
+                hole_height = random.randint(self.min_height, self.max_height)
+                hole_width = random.randint(self.min_width, self.max_width)
+            elif all(
+                [
+                    isinstance(self.min_height, float),
+                    isinstance(self.min_width, float),
+                    isinstance(self.max_height, float),
+                    isinstance(self.max_width, float),
+                ]
+            ):
+                hole_height = int(height * random.uniform(self.min_height, self.max_height))
+                hole_width = int(width * random.uniform(self.min_width, self.max_width))
```

Comment:
I think would be great to add a fix prevent situation when height or width equal to 0 using these lines:
```python
hole_height = max(1, hole_height)
hole_width = max(1, hole_width)
```

These need because, value may be equal to `0.9` for example, but it would be clipped to `0`

Reply:
Even if sometimes height or width becomes 0 (considering usual image sizes, I think that itself is going to be very rare),  I think that is fine because that would return the image as it is. Part of random augmentation I would say.